PA193_mnemonic_Slytherin/mnemonic.py

```diff
@@ -26,15 +25,87 @@
 SEED_LEN = 64
 
 
-def _generate_seed(mnemonic: str, seed_password: str = '') -> bytes:
+class Mnemonic:
+    """Class for mnemonic representation.
+    """
+    
+    def __init__(self, mnemonic: str):
+        """Convert mnemonic phrase to entropy using dictionary to ensure its validity.
+        :raises ValueError: on invalid parameters
+        """
+        if not isinstance(mnemonic, str):
+            raise ValueError('Cannot instantiate mnemonic')
+    
+        words = mnemonic.split()
+        l = len(words)
+        if not l in (12, 15, 18, 21, 24):
+            raise ValueError('Cannot instantiate mnemonic')
+
+        _, word_dict = __get_dictionary()
+        try:
+            indexes = [word_dict[word] for word in words]
+        except KeyError:
+            raise ValueError('Cannot instantiate mnemonic')
+
+        # Concatenate indexes into single 
+        indexes_bin = sum([indexes[-i - 1] << i * 11 for i in reversed(range(l))])
+
+        # Number of bits entropy is shifted by
+        shift = l * 11 // 32
+
+        checksum = indexes_bin & (pow(2, shift) - 1)
+        entropy_bin =  indexes_bin >> shift
+        entropy = entropy_bin.to_bytes((l * 11 - shift) // 8, byteorder='big')
+
+        # Check correctness
+        check = __get_checksum(entropy, shift)
+        if check != checksum:
+            raise ValueError('Cannot instantiate mnemonic')
+    
+        self._entropy = Entropy(entropy)
+        self._str = mnemonic
+
+
+class Entropy:
```

Comment:
If this class would inherit from `bytes`, you don't have to later acess `entropy._bytes`, but can use `entropy` directly.

Reply:
really smart :) , didnt think of this